src/CattyTests/Mocks/URLSessionMock.swift

```diff
@@ -25,31 +25,48 @@ class URLSessionMock: URLSession {
 
     var response: URLResponse?
     var error: Error?
+    var bytesReceived: Int64?
+    var bytesTotal: Int64?
 
-    init(response: URLResponse? = nil, error: Error? = nil) {
+    init(response: URLResponse? = nil, error: Error? = nil, bytesReceived: Int64? = nil, bytesTotal: Int64? = nil) {
         self.response = response
         self.error = error
+        self.bytesReceived = bytesReceived
+        self.bytesTotal = bytesTotal
         super.init()
     }
 
     override func dataTask(with url: URL, completionHandler: @escaping (Data?, URLResponse?, Error?) -> Void) -> URLSessionDataTask {
-        return URLSessionDataTaskMock(completionHandler, response: response, error: error)
+        return URLSessionDataTaskMock(self.bytesReceived, self.bytesTotal, completionHandler, response: response, error: error)
     }
 
     override func dataTask(with request: URLRequest, completionHandler: @escaping (Data?, URLResponse?, Error?) -> Void) -> URLSessionDataTask {
-        return URLSessionDataTaskMock(completionHandler, response: response, error: error)
+        return URLSessionDataTaskMock(self.bytesReceived, self.bytesTotal, completionHandler, response: response, error: error)
     }
 
     class URLSessionDataTaskMock: URLSessionDataTask {
 
+        private var bytesReceived: Int64?
+        private var bytesTotal: Int64?
+
+        override var countOfBytesReceived: Int64 {
+            return bytesReceived!
```

Comment:
Please do not force unwrap this value. You can set a default value of 0 to _bytesReceived_ and _bytesTotal_ (e.g. at the initializer).